### PR TITLE
rename div_euc -> div_euclid, and mod_euc -> rem_euclid

### PR DESCRIPTION
**Pull request by strega-nil**
logic is written up in #49048

Also, update the documentation slightly.

cc @alexcrichton @clarcharr @varkor

### CODE REVIEW

**Comment by clarcharr** [comment]
Thank you for the PR!

File: src/libcore/num/mod.rs [hunk 1]
```diff
@@ -1775,9 +1779,9 @@ assert_eq!((-a).div_euc(-b), 2); // -7 >= -4 * 2
 
 
         doc_comment! {
-            concat!("Calculates the remainder `self mod rhs` by Euclidean division.
+            concat!("Calculates the least nonnegative remainder of `self (mod rhs)`.
 
-In particular, the result `n` satisfies `0 <= n < rhs.abs()`.
```
Comment (dtolnay) [inline src/libcore/num/mod.rs]: I would prefer to keep this inequality in the documentation. This is clearer than "least nonnegative remainder" and "as if by euclidean division algorithm."

File: src/libcore/num/mod.rs [hunk 2]
```diff
@@ -1775,9 +1779,9 @@ assert_eq!((-a).div_euc(-b), 2); // -7 >= -4 * 2
...
+This is done as if by the euclidean division algorithm.
```
Comment (dtolnay) [inline src/libcore/num/mod.rs]:
```suggestion
This is done as if by the Euclidean division algorithm.
```

**Comment by dtolnay** [comment]
Implementation looks good to me. Let's run these names by @rust-lang/libs in case other people have opinions.

Notice there are two changes:

- euc -> euclid
- mod -> rem

File: src/libcore/num/mod.rs (hunk 1 above)
Comment (ubsan) [reply src/libcore/num/mod.rs]: can do

**Pull request merged**